include/swift/AST/Expr.h

```diff
@@ -149,7 +149,7 @@ class alignas(8) Expr {
     IsTypeDefaulted : 1,
     /// Number of comma source locations.
     NumCommas : 32 - 1 - NumExprBits,
-    /// Number of entries in the collection. If this is a DictionaryLiteral,
+    /// Number of entries in the collection. If this is a KeyValueArray,
```

Comment:
I think this refers to `DictionaryExpr` (which is a subclass of `CollectionExpr`).

Reply:
The tuples are probably those given to any [`init(dictionaryLiteral:)`](https://github.com/apple/swift/blob/b6e637e03d012121c59c02aab7915f563401814e/stdlib/public/core/CompilerProtocols.swift#L699-L706) initializer. I don't think `AST/Expr.h` would need to have any knowledge of the `KeyValueArray` type.

Reply:
This is part of the confusion and reason behind the rename. I believe while `AST/Expr.h` has not much to do with concrete types, the comment seems to be referring to the concrete type. Since  this comment is not user facing I am not too worried about it.  You are welcome to change it back. I added you as a collaborator to my branch. Are you able to start the CI? Thanks!

Reply:
I haven't got commit access, so I'm not able to trigger CI testing.

But the [swift-package-manager](https://github.com/apple/swift-package-manager/search?q=DictionaryLiteral&type=Code) didn't compile last time, did it?

You'll either need to:

- use [cross repository testing](https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#cross-repository-testing) with another PR, or
- wait until the compiler supports typealias extensions.

I also wondered if a [generic typealias](https://github.com/apple/swift-evolution/blob/master/proposals/0048-generic-typealias.md) might help, but it didn't seem to (with a recent Swift 4.2 snapshot).